cordova-lib/src/platforms/PlatformApi.js

```diff
@@ -0,0 +1,365 @@
+/**
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+*/
+
+var Q = require('q'),
+    fs = require('fs'),
+    path = require('path'),
+    shell = require('shelljs'),
+    events = require('../events'),
+    util = require('../cordova/util'),
+    superspawn = require('../cordova/superspawn'),
+    platforms = require('./platformsConfig.json'),
+    ConfigParser = require('../configparser/ConfigParser');
+
+var BasePluginHandler = require('../plugman/platforms/PluginHandler');
+var ParserHelper = require('../cordova/metadata/parserhelper/ParserHelper');
+
+// Avoid loading the same platform projects more than once (identified by path)
+var cachedProjects = {};
+
+// A single class that exposes functionality from platform specific files from
```

Comment:
Updated in https://github.com/MSOpenTech/cordova-lib/commit/c78eae57929ed541ac2c2a47a164938433133309